### ISSUE TITLE
Documentation?

### ISSUE THREAD

**Issue by howardjones** [thejameskyle/marionette-wires#53]
I'm taking my 3rd pass at moving an old and horrible php/jquery-soup application to a Marionette-based SPA app, and this project has been very handy! But...

it's cool to be opinionated, but it would be great to have a document that lists those opinions. e.g. why there's a 'common' directory with two-line require/export pairs. That all inter-module communication is done via Radio channels. Nothing fancy, just a simple description of what is going on in the overall design. I also spent a few confused hours last night figuring out why my browserify command-line didn't recognise the code from Wires (the gulpfile didn't look any different from the options I was using, but babelify is attached in the package.json, not the gulpfile).

Just an overview of the structure and design goals would be really useful for newbies like me.

**Comment by inspiredbytes** [comment]
There are a lot ideas incorporated into Marionette Wires. I second @howardjones. I would like to understand the reasons behind the opinions.

I would love to see something like what Brian Mann did with http://www.backbonerails.com/ for his ideas about marionette marionette architecture.

**Comment by zerbons** [comment]
I agree, some documentation would be helpful in getting a better understanding of how the app passes the container to the subapps...

**Comment by keeev** [comment]
Yep, sounds awesome. I try to figure out some stuff since four days right now (in my case I use Jade instead of Handlebars). Like the docs say at the end that reading the sourcecode helps (it does) it would be awesome to have a more detailed Documentation.  

Maybe we can work on something together?

**Comment by ddoronin** [comment]
Hi guys, could you please assist me, I'm trying to understand the magic behind the **gulp** file, but I cannot figure out where is an entry point for babel. I clearly see babel in the package dev-dependencies, I found a browserify() call in the bundler() function, but there is no babel. I don't understand how it works that's why I cannot sleep, eat and code. Give me a hint please :)

**Comment by howardjones** [comment]
Browserify uses the Babelify plugin to do the conversion from ES6 - this is in the package.json, so the browserify task in the gulpfile doesn't make it obvious.

**Comment by ddoronin** [comment]
@howardjones thank you! It's so easy, I just didn't know about [this](https://github.com/substack/browserify-handbook#browserifytransform-field) cool feature.
